workers/unity/Assets/Gdk/Core/Components/Option.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+
+namespace Improbable.Gdk.Core
+{
+    public struct Option<T> : IEquatable<Option<T>>
+    {
+        private bool hasValue;
+        private T value;
+        
+        public Option(T newValue)
+        {
+            hasValue = false;
+            value = default(T);
+            Value = newValue;
+        }
+
+        public BlittableBool HasValue => hasValue;
+
+        public T Value
+        {
+            get
+            {
+                if (!hasValue)
+                {
+                    throw new InvalidOperationException("Called Value on empty Option.");
+                }
+
+                return value;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentException("newValue is null.");
+                }
+
+                hasValue = true;
+                this.value = value;
+            }
+        }
+
+        public void Clear()
+        {
+            hasValue = false;
+            value = default(T);
+        }
+
+        public bool TryGetValue(out T value)
+        {
+            if (!hasValue)
+            {
+                value = default(T);
+                return false;
+            } 
+
+            value = this.value;
+            return true;
+        }
+
+        public T GetValueOrDefault(T defaultValue)
+        {
+            return hasValue ? value : defaultValue;
+        }
+
+        public override bool Equals(object other)
+        {
+            if (other is Option<T>)
```

Comment:
The body of this function can be replaced with: 

```return other is Option<T> && this.Equals((Option<T>)other);```

Reply:
Nice. I love the simplicity.